YouVsVirus/Assets/Scenes/levelsupermarket/EndLevelControllerLevelsupermarket.cs

```diff
@@ -26,4 +26,17 @@ protected override void CummulativeSpriteUpdate()
         // all NPCs show true infection statuts
         CreateHumans.GetComponent<CreatePopulationLevelsupermarket>().CummulativeSpriteUpdate();
     }
+
+    
+    /// <summary>
+    /// Query whether the player is allowed to enter its home.
+    /// In this level, the player is allowed to enter when he has its toilet paper.
+    /// </summary>
+    /// <return> True if and only if the player has the toilet paper </return>
+    public override bool isPlayerAllowedHome(GameObject player)
+    {
+        return player.GetComponent<Player>().hasToiletpaper;
+    }
+
+    // TODO: Why does this endlevelcontroller not know about the toilet paper? Having the toilet paper is a crucial condition for ending the game successfully.
```

Comment:
In my version the player house was responsible for allowing the player to enter or not to enter depending on whether he has toilet paper or not